### PR TITLE
Add Pixel Tests

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR adds screenshot uploading to Percy to the CI. Check out the 4th check at the bottom of this PR!

Transitive overlay doesn't seem to work in the Percy environment, because javascript is disabled (as is recommended by Percy).

In the future, we may want to add some URL based additional snapshots with javascript enabled that will be able to test this sort of thing. I've started out, but the URL parsing isn't quite working.

I've experimentally added a second run using a live server -- this will probably require more manual intervention (updating the date around gtfs service changes, accounting for GBFS changes, etc). If this gets annoying, we can look into hosting an OTP instance with outdated gtfs that will always provide the same results here.

### CODE REVIEW

File: percy/percy.test.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+import puppeteer from 'puppeteer'
+
+const percySnapshot = require('@percy/puppeteer')
+
+const OTP_RR_TEST_CONFIG_PATH = '../percy/har-mock-config.yml'
+const OTP_RR_TEST_JS_CONFIG_PATH = './percy/har-mock-config.js'
+
+const MOCK_SERVER_PORT = 5000
+
+// Puppeteer can take a long time to load, espeically in some ci environments
```
Comment (binh-dam-ibigroup) [inline percy/percy.test.js:14]: Typo "especially"

File: percy/percy.test.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      // headless: false
+    })
+
+    // Fix time
+    browser.on('targetchanged', async (target) => {
+      const targetPage = await target.page()
+      const client = await targetPage.target().createCDPSession()
+      await client.send('Runtime.evaluate', {
+        expression:
+          'Date.now = function() { return 1646835742000; };Date.getTime = function() { return 1646835742000; }'
```
Comment (binh-dam-ibigroup) [inline percy/percy.test.js:89]: Insert space after inner `;`.

File: percy/percy.test.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    console.log(error)
+  }
+  console.log('Closed mock server and headless browser')
+})
+
+// Puppeteer can take a long time to load, espeically in some ci environments
+jest.setTimeout(600000)
+
+/* These fixed routes allow us to test features that the static html screenshots
+ * don't allow us to test. Unfortuantley, they don't currently work
```
Comment (binh-dam-ibigroup) [inline percy/percy.test.js:115]: Typo "unfortunately".

Comment (binh-dam-ibigroup) [reply percy/percy.test.js:115]: Replace this whole comment with a TODO comment to make percy work with transitive.js.

File: percy/percy.test.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    "//span[contains(., 'Sugarloaf Mills - Lindbergh Center')]"
+  )
+  await busRouteButton.click()
+  await page.waitForSelector('#headsign-selector')
+
+  await percySnapshotWithWait(page, 'Route Viewer Showing Route 410')
+
+  // View multiple patterns
+  // Click second option
+  const sugarloafOption = await page.$$eval(
```
Comment (binh-dam-ibigroup) [inline percy/percy.test.js:201]: Maybe rename to `firstPattern` and the other variable to `secondPattern`

**Review by binh-dam-ibigroup — APPROVED** [review]
Thanks for setting this up! See a few formatting comments.

File: percy/percy.test.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply percy/percy.test.js]: Shouldn't have disabled code spell 🤦‍♀️

File: percy/percy.test.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply percy/percy.test.js]: Good call thanks! Refactored in 67e1dc094ba9047bede64664b66a399085257cd0

File: percy/percy.test.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply percy/percy.test.js]: That does feel better thanks! Done in 67e1dc094ba9047bede64664b66a399085257cd0

File: percy/har-mock-config.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
+branding: HAR test
+title: test environment
+homeTimezone: America/Chicago
+
+# Default OTP API
+api:
+  host: http://localhost:9999
+  # host: http://localhost:8001 # For testing against a local OTP instance
```
Comment (daniel-heppner-ibigroup) [inline percy/har-mock-config.yml:8]: Remove this line

**Review by daniel-heppner-ibigroup — APPROVED** [review]

File: percy/har-mock-config.yml (hunk 5 above)
Comment (miles-grant-ibigroup) [reply percy/har-mock-config.yml]: Good call! Removed in 46a7627572c613a1abb4e1e13e1a481c6e7a3245

**Pull request merged**